src/rpc/core_rpc_server.cpp

```diff
@@ -1038,6 +1154,11 @@ namespace cryptonote
   bool core_rpc_server::on_submitblock(const COMMAND_RPC_SUBMITBLOCK::request& req, COMMAND_RPC_SUBMITBLOCK::response& res, epee::json_rpc::error& error_resp)
   {
     PERF_TIMER(on_submitblock);
+    if (m_should_use_bootstrap_daemon)
```

Comment:
UB (see above), locking the mutex is required.

Reply:
nope